tools/build_script_generator/cmake/module.lb

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017-2018, Fabian Greif
+# Copyright (c) 2017-2018, Niklas Hauser
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+import os
+
+
+def init(module):
+    module.name = "build.cmake"
+    module.description = """CMake Build Script Generator
+
+Generates a CMakeLists.txt file to build the modm library.
+"""
+
+
+def prepare(module, options):
+    _, default_project_name = os.path.split(os.getcwd())
+
+    module.add_option(
+        BooleanOption(name="include_demo_project", default=False,
+                      description="Include a demo main file to get a full compilable project."))
+    module.add_option(
+        BooleanOption(name="include_sconstruct", default=True,
+                      description="Generate modm specific SConstruct file."))
+    module.add_option(
+        BooleanOption(name="is_unittest", default=False,
+                      description="Generate SConstruct file for modm unittests."))
+
```

Comment:
Lets remove non-essential option incl. the whole demo project thing, which was more of a development thing.
The XPCC protocol stuff is only used RCA-internally anyways, it makes no sense to port it to CMake.
So, lets remove all options you've not ported now: 
- `include_demo_project`
- `include_sconstruct`
- `is_unittest`
- `xpcc.*` (you renamed this to `modm.*`)
- `image.*`
- `info.*`

Reply:
PS: The `xpcc.*` options are not a typo, there exists a XPCC (Cross-Platform Component Communication) protocol, which does RPC-style communications over CAN. It's used only by the RCA and it's basically not documented 🙄. This library sort of grew around it.

Reply:
Done.